src/System.Management.Automation/engine/hostifaces/LocalPipeline.cs

```diff
@@ -247,54 +248,35 @@ private void SetupInvokeThread(Thread invokeThread, bool changeName)
             }
         }
 
-#if !CORECLR
         /// <summary>
-        /// Stack Reserve setting for pipeline threads
+        /// Set stack size for local pipeline threads.
+        /// Default is 10MB.
+        /// Default 10MB may be too large if we use many local pipelines for paralleling.PipelineMaxStackSizeMB
         /// </summary>
-        internal static int MaxStack
-        {
-            get
-            {
-                int i = ReadRegistryInt("PipelineMaxStackSizeMB", 10);
-                if (i < 10)
-                    i = 10; // minimum 10MB
-                else if (i > 100)
-                    i = 100; // maximum 100MB
-                return i * 1000000;
-            }
-        }
+        internal static int StackSize => GetPipelineMaxStackSizeOption(10);
```

Comment:
If we don't cache the stack size, then this static property seems not necessary as it only adds one more redirection to call `GetPipelineMaxStackSizeOption`.